src/Service/RegisteredMethodManager.php

```diff
@@ -61,4 +62,33 @@ public function registerForMember(Member $member, MethodInterface $method, $data
 
         $this->extend('onRegisterMethod', $member, $method);
     }
+
+    /**
+     * Delete a registration for the given method from the given member, provided it exists. This will also remove a
+     * registered back-up method if it will leave the member with only the back-up method remaing
+     *
+     * @param Member&MemberExtension $member
+     * @param MethodInterface $method
+     * @return bool Returns false if the given method is not registered for the member
+     */
+    public function deleteFromMember(Member $member, MethodInterface $method): bool
+    {
+        $method = $this->getFromMember($member, $method);
+
+        if (!$method) {
+            return false;
+        }
+
+        $method->delete();
+
+        // If there is only one method remaining, and that's the configured "backup" method - then delete that too
+        if ($member->RegisteredMFAMethods()->count() === 1
+            && $member->RegisteredMFAMethods()->first()->MethodClassName
+                === Config::inst()->get(MethodRegistry::class, 'default_backup_method')
+        ) {
+            $member->RegisteredMFAMethods()->first()->delete();
```

Comment:
This is the second call to `RegisteredMFAMethods()->first` - I feel like it would be better to assign the earlier result to a variable, the calls are not cached as I recall it.

Reply:
Done :)